frontend/src/lib/components/AddInsightsToDashboard/addInsightsToDashboardModalLogic.ts

```diff
@@ -0,0 +1,101 @@
+import FuseClass from 'fuse.js'
+import { actions, connect, kea, key, listeners, path, props, reducers, selectors } from 'kea'
+import { eventUsageLogic } from 'lib/utils/eventUsageLogic'
+import { dashboardLogic } from 'scenes/dashboard/dashboardLogic'
+import { savedInsightsLogic } from 'scenes/saved-insights/savedInsightsLogic'
+
+import { dashboardsModel } from '~/models/dashboardsModel'
+import { DashboardType, InsightModel, InsightShortId } from '~/types'
+
+import type { addInsightsToDashboardModalLogicType } from './addInsightsToDashboardModalLogicType'
+
+export interface AddInsightsToDashboardModalLogicProps {
+    dashboard: DashboardType
+}
+
+export interface Fuse extends FuseClass<any> {}
+
+export const addInsightsToDashboardModalLogic = kea<addInsightsToDashboardModalLogicType>([
+    props({} as AddInsightsToDashboardModalLogicProps),
+    key(({ dashboard }) => {
+        if (!dashboard.id) {
+            throw Error('must provide a dashboard id')
+        }
+        return dashboard.id
+    }),
+    path(['lib', 'components', 'AddInsightsToDashboard', 'saveInsightsToDashboardModalLogic']),
```

Comment:
```suggestion
    path((key) => ['lib', 'components', 'AddInsightsToDashboard', 'addInsightsToDashboardModalLogic', key]),
```
Path seems to have changed and it would be good to include the key here :)